### PR TITLE
Unit tests for hs_core view functions

### PR DESCRIPTION
**Pull request by pkdash**
Ready for review

Refer to Issue #1940 for a list of view functions for which unit tests have been written as part of this PR.

With these new set of unit tests the coverage seems to go up by 1% point.

### CODE REVIEW

**Comment by pkdash** [comment]
@hyi Once you are done with your quota work, can you review this PR?

File: hs_app_netCDF/tests/views/test_create_netcdf_resource.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        url = reverse('create_resource')
+        request = self.factory.post(url, data=post_data)
+        request.user = self.user
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+
+        response = create_resource(request)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+        res_id = response.url.split('/')[2]
+        self.assertEqual(BaseResource.objects.filter(short_id=res_id).exists(), True)
```
Comment (hyi) [inline hs_app_netCDF/tests/views/test_create_netcdf_resource.py]: You can use assertTrue() instead

File: hs_app_netCDF/tests/views/test_create_netcdf_resource.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        # there should be 2 files (nc and txt) - successful metadata extraction
+        self.assertEqual(ResourceFile.objects.count(), 2)
+
+        hydroshare.delete_resource(res_id)
+
```
Comment (hyi) [inline hs_app_netCDF/tests/views/test_create_netcdf_resource.py]: add ```self.assertEqual(BaseResource.objects.count(), 0)``` after delete_resource() call

File: hs_app_netCDF/tests/views/test_create_netcdf_resource.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        url = reverse('create_resource')
+        request = self.factory.post(url, data=post_data)
+        request.user = self.user
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+
+        response = create_resource(request)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        # this should take back to create_resource.html page as resource was not created
+        self.assertIn('<!doctype html>', response.content)
```
Comment (hyi) [inline hs_app_netCDF/tests/views/test_create_netcdf_resource.py]: Perhaps in addition to this assertIn(), adding a more create-resource page specific assertIn() is better such as ```self.assertIn('create-resource', response.content)```?

File: hs_app_timeseries/tests/views/test_create_timeseries_resource.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        # tested uses messaging middleware
+        setattr(request, 'session', 'session')
+        messages = FallbackStorage(request)
+        setattr(request, '_messages', messages)
+
+    def _add_session_to_request(self, request):
+        """Annotate a request object with a session"""
+        middleware = SessionMiddleware()
+        middleware.process_request(request)
+        request.session.save()
```
Comment (hyi) [inline hs_app_timeseries/tests/views/test_create_timeseries_resource.py]: Above two internal method ```_set_request_message_attributes``` and ```_add_session_to_request``` should be consolidated into common methods that can be used by view tests of different resource types to eliminate duplications.

File: hs_app_timeseries/tests/views/test_create_timeseries_resource.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+        url = reverse('create_resource')
+        request = self.factory.post(url, data=post_data)
+        request.user = self.user
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+
+        response = create_resource(request)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+        res_id = response.url.split('/')[2]
+        self.assertEqual(BaseResource.objects.filter(short_id=res_id).exists(), True)
```
Comment (hyi) [inline hs_app_timeseries/tests/views/test_create_timeseries_resource.py]: change to to assertTrue(). You can search through all occurrences of such and make changes to all of them

File: hs_composite_resource/tests/views/test_create_composite_resource.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        url = reverse('create_resource')
+        request = self.factory.post(url, data=post_data)
+        request.user = self.user
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+
+        response = create_resource(request)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+        res_id = response.url.split('/')[2]
+        self.assertEqual(BaseResource.objects.filter(short_id=res_id).exists(), True)
```
Comment (hyi) [inline hs_composite_resource/tests/views/test_create_composite_resource.py]: assertTrue

File: hs_composite_resource/tests/views/test_create_composite_resource.py [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+        # tested uses messaging middleware
+        setattr(request, 'session', 'session')
+        messages = FallbackStorage(request)
+        setattr(request, '_messages', messages)
+
+    def _add_session_to_request(self, request):
+        """Annotate a request object with a session"""
+        middleware = SessionMiddleware()
+        middleware.process_request(request)
+        request.session.save()
```
Comment (hyi) [inline hs_composite_resource/tests/views/test_create_composite_resource.py]: remove duplicate and use common methods instead

File: hs_core/tests/api/views/test_add_delete_resource_files.py [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+        request = self.factory.post(url, data=post_data)
+        request.user = self.user
+        # make it a ajax request
+        request.META['HTTP_X_REQUESTED_WITH'] = 'XMLHttpRequest'
+        self._set_request_message_attributes(request)
+        response = add_files_to_resource(request, shortkey=self.gen_res.short_id)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        # there should be 1 file
+        self.assertEqual(self.gen_res.files.count(), 1)
+        hydroshare.delete_resource(self.gen_res.short_id)
```
Comment (hyi) [inline hs_core/tests/api/views/test_add_delete_resource_files.py]: add assert to make sure resource is deleted.

File: hs_core/tests/api/views/test_add_delete_resource_files.py [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+        # tested uses messaging middleware
+        setattr(request, 'session', 'session')
+        messages = FallbackStorage(request)
+        setattr(request, '_messages', messages)
+
+    def _add_session_to_request(self, request):
+        """Annotate a request object with a session"""
+        middleware = SessionMiddleware()
+        middleware.process_request(request)
+        request.session.save()
```
Comment (hyi) [inline hs_core/tests/api/views/test_add_delete_resource_files.py]: remove duplicate methods and use common methods instead

File: hs_core/tests/api/views/test_create_resource.py [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+        url = reverse('create_resource')
+        request = self.factory.post(url, data=post_data)
+        request.user = self.user
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+
+        response = create_resource(request)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+        res_id = response.url.split('/')[2]
+        self.assertEqual(BaseResource.objects.filter(short_id=res_id).exists(), True)
```
Comment (hyi) [inline hs_core/tests/api/views/test_create_resource.py]: assertTrue

File: hs_core/tests/api/views/test_create_resource.py [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+        # tested uses messaging middleware
+        setattr(request, 'session', 'session')
+        messages = FallbackStorage(request)
+        setattr(request, '_messages', messages)
+
+    def _add_session_to_request(self, request):
+        """Annotate a request object with a session"""
+        middleware = SessionMiddleware()
+        middleware.process_request(request)
+        request.session.save()
```
Comment (hyi) [inline hs_core/tests/api/views/test_create_resource.py]: use common method instead

File: hs_core/tests/api/views/test_delete_resource.py [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+        # tested uses messaging middleware
+        setattr(request, 'session', 'session')
+        messages = FallbackStorage(request)
+        setattr(request, '_messages', messages)
+
+    def _add_session_to_request(self, request):
+        """Annotate a request object with a session"""
+        middleware = SessionMiddleware()
+        middleware.process_request(request)
+        request.session.save()
```
Comment (hyi) [inline hs_core/tests/api/views/test_delete_resource.py]: use common methods

File: hs_core/tests/api/views/test_group_CRUD.py [hunk 13]
```diff
@@ -0,0 +1,312 @@
...
+        expected_new_group_id = Group.objects.all().order_by("-id").first().id + 1
+        request.META['HTTP_REFERER'] = '/group/{}'.format(expected_new_group_id)
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+        response = create_user_group(request)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+        self.assertEqual(response['Location'], request.META['HTTP_REFERER'])
+        flag_messages = get_messages(request)
+        success_messages = [m for m in flag_messages if m.tags == 'success']
+        self.assertNotEqual(len(success_messages), 0)
```
Comment (hyi) [inline hs_core/tests/api/views/test_group_CRUD.py]: assertGreater() is better here?

File: hs_core/tests/api/views/test_group_CRUD.py [hunk 14]
```diff
@@ -0,0 +1,312 @@
...
+        expected_new_group_id = Group.objects.all().order_by("-id").first().id + 1
+        request.META['HTTP_REFERER'] = '/group/{}'.format(expected_new_group_id - 1)
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+        response = create_user_group(request)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+        self.assertNotEqual(response['Location'], '/group/{}'.format(expected_new_group_id))
+        flag_messages = get_messages(request)
+        err_messages = [m for m in flag_messages if m.tags == 'error']
+        self.assertNotEqual(len(err_messages), 0)
```
Comment (hyi) [inline hs_core/tests/api/views/test_group_CRUD.py]: assertGreater() is better here?

File: hs_core/tests/api/views/test_group_CRUD.py [hunk 15]
```diff
@@ -0,0 +1,312 @@
...
+        request.user = self.john
+        request.META['HTTP_REFERER'] = '/group/{}'.format(new_group.id)
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+        response = update_user_group(request, group_id=new_group.id)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+        self.assertEqual(response['Location'], request.META['HTTP_REFERER'])
+        flag_messages = get_messages(request)
+        success_messages = [m for m in flag_messages if m.tags == 'success']
+        self.assertNotEqual(len(success_messages), 0)
```
Comment (hyi) [inline hs_core/tests/api/views/test_group_CRUD.py]: assertGreater() is better?

File: hs_core/tests/api/views/test_group_CRUD.py [hunk 16]
```diff
@@ -0,0 +1,312 @@
...
+        self.assertEqual(Group.objects.count(), 1)
+        # let john create a new group
+        new_group = self.john.uaccess.create_group(title='Test Group',
+                                                   description='This is a great group',
+                                                   purpose='To have fun',
+                                                   auto_approve=True)
+
+        # there should be 2 groups at this point
+        self.assertEqual(Group.objects.count(), 2)
+        self.assertEqual(new_group.gaccess.active, True)
```
Comment (hyi) [inline hs_core/tests/api/views/test_group_CRUD.py]: assertTrue

File: hs_core/tests/api/views/test_group_CRUD.py [hunk 17]
```diff
@@ -0,0 +1,312 @@
...
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+        response = delete_user_group(request, group_id=new_group.id)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+        self.assertEqual(response['Location'], request.META['HTTP_REFERER'])
+        flag_messages = get_messages(request)
+        success_messages = [m for m in flag_messages if m.tags == 'success']
+        self.assertNotEqual(len(success_messages), 0)
+        new_group.gaccess.refresh_from_db()
+        self.assertEqual(new_group.gaccess.active, False)
```
Comment (hyi) [inline hs_core/tests/api/views/test_group_CRUD.py]: assertFalse

File: hs_core/tests/api/views/test_group_CRUD.py [hunk 18]
```diff
@@ -0,0 +1,312 @@
...
+        self._add_session_to_request(request)
+        response = delete_user_group(request, group_id=new_group.id)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+        self.assertEqual(response['Location'], request.META['HTTP_REFERER'])
+        flag_messages = get_messages(request)
+        err_messages = [m for m in flag_messages if m.tags == 'error']
+        self.assertNotEqual(len(err_messages), 0)
+        new_group.gaccess.refresh_from_db()
+        # group should still be active
+        self.assertEqual(new_group.gaccess.active, True)
```
Comment (hyi) [inline hs_core/tests/api/views/test_group_CRUD.py]: assertTrue

File: hs_core/tests/api/views/test_group_CRUD.py [hunk 19]
```diff
@@ -0,0 +1,312 @@
...
+        self._add_session_to_request(request)
+        response = restore_user_group(request, group_id=new_group.id)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+        self.assertEqual(response['Location'], request.META['HTTP_REFERER'])
+        flag_messages = get_messages(request)
+        success_messages = [m for m in flag_messages if m.tags == 'success']
+        self.assertNotEqual(len(success_messages), 0)
+        new_group.gaccess.refresh_from_db()
+        # group should be active
+        self.assertEqual(new_group.gaccess.active, True)
```
Comment (hyi) [inline hs_core/tests/api/views/test_group_CRUD.py]: assertTrue

File: hs_core/tests/api/views/test_group_CRUD.py [hunk 20]
```diff
@@ -0,0 +1,312 @@
...
+        self.assertEqual(Group.objects.count(), 1)
+        # let john create a new group
+        new_group = self.john.uaccess.create_group(title='Test Group',
+                                                   description='This is a great group',
+                                                   purpose='To have fun',
+                                                   auto_approve=True)
+
+        # there should be 2 groups at this point
+        self.assertEqual(Group.objects.count(), 2)
+        self.assertEqual(new_group.gaccess.active, True)
```
Comment (hyi) [inline hs_core/tests/api/views/test_group_CRUD.py]: assertTrue

File: hs_core/tests/api/views/test_group_CRUD.py [hunk 21]
```diff
@@ -0,0 +1,312 @@
...
+        self._add_session_to_request(request)
+        response = restore_user_group(request, group_id=new_group.id)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+        self.assertEqual(response['Location'], request.META['HTTP_REFERER'])
+        flag_messages = get_messages(request)
+        err_messages = [m for m in flag_messages if m.tags == 'error']
+        self.assertNotEqual(len(err_messages), 0)
+        new_group.gaccess.refresh_from_db()
+        # group should still be inactive
+        self.assertEqual(new_group.gaccess.active, False)
```
Comment (hyi) [inline hs_core/tests/api/views/test_group_CRUD.py]: assertFalse

File: hs_core/tests/api/views/test_group_CRUD.py [hunk 22]
```diff
@@ -0,0 +1,312 @@
...
+        # tested uses messaging middleware
+        setattr(request, 'session', 'session')
+        messages = FallbackStorage(request)
+        setattr(request, '_messages', messages)
+
+    def _add_session_to_request(self, request):
+        """Annotate a request object with a session"""
+        middleware = SessionMiddleware()
+        middleware.process_request(request)
+        request.session.save()
```
Comment (hyi) [inline hs_core/tests/api/views/test_group_CRUD.py]: use common methods

File: hs_core/tests/api/views/test_group_share.py [hunk 23]
```diff
@@ -0,0 +1,451 @@
...
+    def _check_membership_request(self, invitation=True):
+
+        user_id = None
+        if invitation:
+            user_id = self.mike.id
+            user = self.john
+        else:
+            user = self.mike
+
+        if user_id is not None:
```
Comment (hyi) [inline hs_core/tests/api/views/test_group_share.py]: simplify it to ```if user_id:```

File: hs_core/tests/api/views/test_group_share.py [hunk 24]
```diff
@@ -0,0 +1,451 @@
...
+            url_params = {'group_id': self.test_group.id}
+
+        url = reverse('make_group_membership_request', kwargs=url_params)
+        request = self.factory.post(url, data={})
+        request.user = user
+        request.META['HTTP_REFERER'] = '/group/{}'.format(self.test_group.id)
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+
+        if user_id is not None:
```
Comment (hyi) [inline hs_core/tests/api/views/test_group_share.py]: if user_id:

File: hs_core/tests/api/views/test_group_share.py [hunk 25]
```diff
@@ -0,0 +1,451 @@
...
+        # tested uses messaging middleware
+        setattr(request, 'session', 'session')
+        messages = FallbackStorage(request)
+        setattr(request, '_messages', messages)
+
+    def _add_session_to_request(self, request):
+        """Annotate a request object with a session"""
+        middleware = SessionMiddleware()
+        middleware.process_request(request)
+        request.session.save()
```
Comment (hyi) [inline hs_core/tests/api/views/test_group_share.py]: use common methods

File: hs_core/tests/api/views/test_metadata_CRUD.py [hunk 26]
```diff
@@ -0,0 +1,143 @@
...
+        # tested uses messaging middleware
+        setattr(request, 'session', 'session')
+        messages = FallbackStorage(request)
+        setattr(request, '_messages', messages)
+
+    def _add_session_to_request(self, request):
+        """Annotate a request object with a session"""
+        middleware = SessionMiddleware()
+        middleware.process_request(request)
+        request.session.save()
```
Comment (hyi) [inline hs_core/tests/api/views/test_metadata_CRUD.py]: use common methods

File: hs_core/tests/api/views/test_multiple_file_upload_by_resource_type.py [hunk 27]
```diff
@@ -0,0 +1,97 @@
...
+        resp_json = self._make_request("GeographicFeatureResource")
+        self.assertEqual(resp_json['allow_multiple_file'], True)
+
+        # test for ModelProgramResource
+        resp_json = self._make_request("ModelProgramResource")
+        self.assertEqual(resp_json['allow_multiple_file'], True)
+
+        # test for ModelInstanceResource
+        resp_json = self._make_request("ModelInstanceResource")
+        self.assertEqual(resp_json['allow_multiple_file'], True)
```
Comment (hyi) [inline hs_core/tests/api/views/test_multiple_file_upload_by_resource_type.py]: change to assertTrue() instead for all these similar occurrences.

File: hs_core/tests/api/views/test_multiple_file_upload_by_resource_type.py [hunk 28]
```diff
@@ -0,0 +1,97 @@
...
+        resp_json = self._make_request("ScriptResource")
+        self.assertEqual(resp_json['allow_multiple_file'], True)
+
+        # test for SWATModelInstanceResource
+        resp_json = self._make_request("SWATModelInstanceResource")
+        self.assertEqual(resp_json['allow_multiple_file'], True)
+
+        # test for ToolResource
+        resp_json = self._make_request("ToolResource")
+        self.assertEqual(resp_json['allow_multiple_file'], False)
```
Comment (hyi) [inline hs_core/tests/api/views/test_multiple_file_upload_by_resource_type.py]: assertFalse

File: hs_core/tests/api/views/test_replicate_bag_to_user_zone.py [hunk 29]
```diff
@@ -0,0 +1,57 @@
...
+        self.password = 'jhmypassword'
+        self.user = hydroshare.create_account(
+            'john@gmail.com',
+            username=self.username,
+            first_name='John',
+            last_name='Clarson',
+            superuser=False,
+            password=self.password,
+            groups=[]
+        )
```
Comment (hyi) [inline hs_core/tests/api/views/test_replicate_bag_to_user_zone.py]: I think ```super(TestUserZoneIRODSFederation, self).create_irods_user_in_user_zone()``` needs to be called to ensure the user has a corresponding user account in user zone. Refer to https://github.com/hydroshare/hydroshare/blob/develop/hs_core/tests/api/native/test_irods_user_zone_federation.py for details.

File: hs_core/tests/api/views/test_set_resource_flag.py [hunk 30]
```diff
@@ -0,0 +1,292 @@
...
+        url = reverse('set_resource_flag', kwargs=url_params)
+        request = self.factory.post(url, data=post_data)
+        request.user = self.user
+
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+        set_resource_flag(request, shortkey=self.gen_res_one.short_id)
+        # check that the resource is still not public
+        self.gen_res_one.raccess.refresh_from_db()
+        self.assertEqual(self.gen_res_one.raccess.public, False)
```
Comment (hyi) [inline hs_core/tests/api/views/test_set_resource_flag.py]: assertFalse

File: hs_core/tests/api/views/test_set_resource_flag.py [hunk 31]
```diff
@@ -0,0 +1,292 @@
...
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+        response = set_resource_flag(request, shortkey=self.gen_res_two.short_id)
+        self.assertEqual(response.status_code, status.HTTP_202_ACCEPTED)
+        flag_messages = get_messages(request)
+        err_messages = [m for m in flag_messages if m.tags == 'error']
+        self.assertEqual(len(err_messages), 0)
+        # check that the resource is public now
+        self.gen_res_two.raccess.refresh_from_db()
+        self.assertEqual(self.gen_res_two.raccess.public, True)
```
Comment (hyi) [inline hs_core/tests/api/views/test_set_resource_flag.py]: assertTrue

File: hs_core/tests/api/views/test_set_resource_flag.py [hunk 32]
```diff
@@ -0,0 +1,292 @@
...
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+        response = set_resource_flag(request, shortkey=self.gen_res_one.short_id)
+        self.assertEqual(response.status_code, status.HTTP_202_ACCEPTED)
+        flag_messages = get_messages(request)
+        err_messages = [m for m in flag_messages if m.tags == 'error']
+        self.assertEqual(len(err_messages), 0)
+        # check that the resource is private now
+        self.gen_res_one.raccess.refresh_from_db()
+        self.assertEqual(self.gen_res_one.raccess.public, False)
```
Comment (hyi) [inline hs_core/tests/api/views/test_set_resource_flag.py]: assertFalse

File: hs_core/tests/api/views/test_set_resource_flag.py [hunk 33]
```diff
@@ -0,0 +1,292 @@
...
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+        response = set_resource_flag(request, shortkey=self.gen_res_two.short_id)
+        self.assertEqual(response.status_code, status.HTTP_202_ACCEPTED)
+        flag_messages = get_messages(request)
+        err_messages = [m for m in flag_messages if m.tags == 'error']
+        self.assertEqual(len(err_messages), 0)
+        # check that the resource is discoverable now
+        self.gen_res_two.raccess.refresh_from_db()
+        self.assertEqual(self.gen_res_two.raccess.discoverable, True)
```
Comment (hyi) [inline hs_core/tests/api/views/test_set_resource_flag.py]: assertTrue

File: hs_core/tests/api/views/test_set_resource_flag.py [hunk 34]
```diff
@@ -0,0 +1,292 @@
...
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+        response = set_resource_flag(request, shortkey=self.gen_res_one.short_id)
+        self.assertEqual(response.status_code, status.HTTP_202_ACCEPTED)
+        flag_messages = get_messages(request)
+        err_messages = [m for m in flag_messages if m.tags == 'error']
+        self.assertEqual(len(err_messages), 0)
+        # check that the resource is shareable now
+        self.gen_res_one.raccess.refresh_from_db()
+        self.assertEqual(self.gen_res_one.raccess.shareable, True)
```
Comment (hyi) [inline hs_core/tests/api/views/test_set_resource_flag.py]: assertTrue

File: hs_core/tests/api/views/test_set_resource_flag.py [hunk 35]
```diff
@@ -0,0 +1,292 @@
...
+        # clean up
+        hydroshare.delete_resource(self.gen_res_one.short_id)
+        hydroshare.delete_resource(self.gen_res_two.short_id)
+
+    def test_set_resource_flag_make_not_shareable(self):
+        # here we are testing the set_resource_flag view function to make a resource not shareable
+
+        # test that the resource is  shareable
+        self.assertEqual(self.gen_res_one.raccess.shareable, True)
```
Comment (hyi) [inline hs_core/tests/api/views/test_set_resource_flag.py]: assertTrue

File: hs_core/tests/api/views/test_set_resource_flag.py [hunk 36]
```diff
@@ -0,0 +1,292 @@
...
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+        response = set_resource_flag(request, shortkey=self.gen_res_one.short_id)
+        self.assertEqual(response.status_code, status.HTTP_202_ACCEPTED)
+        flag_messages = get_messages(request)
+        err_messages = [m for m in flag_messages if m.tags == 'error']
+        self.assertEqual(len(err_messages), 0)
+        # check that the resource is not shareable now
+        self.gen_res_one.raccess.refresh_from_db()
+        self.assertEqual(self.gen_res_one.raccess.shareable, False)
```
Comment (hyi) [inline hs_core/tests/api/views/test_set_resource_flag.py]: assertFalse

File: hs_app_netCDF/tests/views/test_create_netcdf_resource.py [hunk 37]
```diff
@@ -0,0 +1,135 @@
...
+        if os.path.exists(self.temp_dir):
+            shutil.rmtree(self.temp_dir)
+
+    def test_create_resource(self):
+        # here we are testing the create_resource view function
+
+        # test with no file upload
+        post_data = {'resource-type': 'NetcdfResource',
+                     'title': 'Test NetCDF Resource Creation',
+                     'irods_federated': 'true'
```
Comment (hyi) [inline hs_app_netCDF/tests/views/test_create_netcdf_resource.py]: should this 'irods_federated' set to 'true' or not?

File: hs_geo_raster_resource/tests/views/test_create_raster_resource.py [hunk 38]
```diff
@@ -0,0 +1,137 @@
...
+        if os.path.exists(self.temp_dir):
+            shutil.rmtree(self.temp_dir)
+
+    def test_create_resource(self):
+        # here we are testing the create_resource view function
+
+        # test with no file upload
+        post_data = {'resource-type': 'RasterResource',
+                     'title': 'Test Raster Resource Creation',
+                     'irods_federated': 'true'
```
Comment (hyi) [inline hs_geo_raster_resource/tests/views/test_create_raster_resource.py]: should this 'irods_federated' set to 'true' or not?

File: hs_geo_raster_resource/tests/views/test_create_raster_resource.py [hunk 39]
```diff
@@ -0,0 +1,137 @@
...
+        res_id = response.url.split('/')[2]
+        self.assertEqual(BaseResource.objects.filter(short_id=res_id).exists(), True)
+        hydroshare.delete_resource(res_id)
+        self.assertEqual(BaseResource.objects.count(), 0)
+
+        # test with file upload
+        self.assertEqual(ResourceFile.objects.count(), 0)
+        post_data = {'resource-type': 'RasterResource',
+                     'title': 'Test Raster Resource Creation',
+                     'irods_federated': 'true',
```
Comment (hyi) [inline hs_geo_raster_resource/tests/views/test_create_raster_resource.py]: should this 'irods_federated' set to 'true' or not?

File: hs_geo_raster_resource/tests/views/test_create_raster_resource.py [hunk 40]
```diff
@@ -0,0 +1,137 @@
...
+        url = reverse('create_resource')
+        request = self.factory.post(url, data=post_data)
+        request.user = self.user
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+
+        response = create_resource(request)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+        res_id = response.url.split('/')[2]
+        self.assertEqual(BaseResource.objects.filter(short_id=res_id).exists(), True)
```
Comment (hyi) [inline hs_geo_raster_resource/tests/views/test_create_raster_resource.py]: assertTrue

File: hs_geo_raster_resource/tests/views/test_create_raster_resource.py [hunk 41]
```diff
@@ -0,0 +1,137 @@
...
+        # tested uses messaging middleware
+        setattr(request, 'session', 'session')
+        messages = FallbackStorage(request)
+        setattr(request, '_messages', messages)
+
+    def _add_session_to_request(self, request):
+        """Annotate a request object with a session"""
+        middleware = SessionMiddleware()
+        middleware.process_request(request)
+        request.session.save()
```
Comment (hyi) [inline hs_geo_raster_resource/tests/views/test_create_raster_resource.py]: use common methods instead

File: hs_app_netCDF/tests/views/test_create_netcdf_resource.py (hunk 2 above)
Comment (pkdash) [inline hs_app_netCDF/tests/views/test_create_netcdf_resource.py]: That would be necessary only if the intent is also to test the ```delete_resource()``` function.

File: hs_core/tests/api/views/test_group_share.py (hunk 23 above)
Comment (pkdash) [inline hs_core/tests/api/views/test_group_share.py]: There is a difference what I have and what you are suggesting. The conditional expression you proposing works for ```None``` as well as ```True/False```.  The expression that I have only cares for ```None``` and more explicit condition.

File: hs_app_netCDF/tests/views/test_create_netcdf_resource.py [hunk 42]
```diff
@@ -0,0 +1,135 @@
...
+        url = reverse('create_resource')
+        request = self.factory.post(url, data=post_data)
+        request.user = self.user
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+
+        response = create_resource(request)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+        res_id = response.url.split('/')[2]
+        self.assertEqual(BaseResource.objects.filter(short_id=res_id).exists(), True)
```
Comment (pkdash) [inline hs_app_netCDF/tests/views/test_create_netcdf_resource.py]: I can use that. But not sure why change. I think when the test fails with assertEqual() it shows more information related failure compared to assertTrue().

Comment (hyi) [inline hs_app_netCDF/tests/views/test_create_netcdf_resource.py]: assertTrue() is simpler and right to the point you are testing here. You should be able to put in more detailed message if it fails to provide more customized message when assert fails. But if you think assertEqual(x, True) works better for you, it is fine with me to keep it the way it is.

**Comment by pkdash** [comment]
@mjstealey It seems the PR build is failing due to timeout.

**Comment by mjstealey** [comment]
The following one failed due failed tests, pep8, etc.

http://ci.hydroshare.org:8080/job/hydroshare-pull-requests/528/ <http://ci.hydroshare.org:8080/job/hydroshare-pull-requests/528/>

**Comment by pkdash** [comment]
@hyi I have refactored the code now to avoid code duplication. I have kept the ```assertEqual()``` where one possibly could use either ```assertTrue()``` or ```assertFalse()```. But I prefer ```assertEqual()```.  See if you can +1 with the changes I have made now.

File: hs_core/tests/api/views/test_replicate_bag_to_user_zone.py (hunk 29 above)
Comment (hyi) [inline hs_core/tests/api/views/test_replicate_bag_to_user_zone.py]: @pkdash It appears this comment has not been addressed.

**Comment by hyi** [comment]
@pkdash I left one further comment regarding calling super().create_irods_user_in_user_zone(). After you address that comment, I will +1.

File: hs_core/tests/api/views/test_replicate_bag_to_user_zone.py (hunk 29 above)
Comment (pkdash) [inline hs_core/tests/api/views/test_replicate_bag_to_user_zone.py]: @hyi Yeah somehow did not see this comment earlier. No have fixed it.

**Comment by hyi** [comment]
+1 then

**Pull request merged**